packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
+	next: (await import("../templates/next/c3")).default,
+	nuxt: (await import("../templates/nuxt/c3")).default,
+	qwik: (await import("../templates/qwik/c3")).default,
+	react: (await import("../templates/react/c3")).default,
+	remix: (await import("../templates/remix/c3")).default,
+	solid: (await import("../templates/solid/c3")).default,
+	svelte: (await import("../templates/svelte/c3")).default,
+	vue: (await import("../templates/vue/c3")).default,
+});
+
+export const getTemplateMap = async () => {
+	return {
+		"hello-world": await import("../templates/hello-world/c3.json"),
+		// Dummy record
+		webFramework: { displayName: "Website or web app" },
+		common: await import("../templates/common/c3.json"),
+		scheduled: await import("../templates/scheduled/c3.json"),
+		queues: await import("../templates/queues/c3.json"),
+		chatgptPlugin: await import("../templates/chatgptPlugin/c3.json"),
+		openapi: await import("../templates/openapi/c3.json"),
+		"remote-template": { displayName: "From a remote git repository" },
+		"pre-existing": (await import("../templates/pre-existing/c3")).default,
+	} as unknown as Record<string, TemplateConfig>;
 };
+
+export const selectTemplate = async (args: Partial<C3Args>) => {
+	// If not specified, attempt to infer the `type` argument from other flags
+	if (!args.type) {
+		if (args.framework) {
+			args.type = "webFramework";
+		} else if (args.existingScript) {
+			args.type = "pre-existing";
+		} else if (args.template) {
+			args.type = "remote-template";
+		}
+	}
+
+	const templateMap = await getTemplateMap();
+
+	const templateOptions = Object.entries(templateMap).map(
+		([value, { displayName, hidden }]) => ({
+			value,
+			label: displayName,
+			hidden,
+		})
+	);
+
+	const type = await processArgument<string>(args, "type", {
+		type: "select",
+		question: "What type of application do you want to create?",
+		label: "type",
+		options: templateOptions,
+		defaultValue: C3_DEFAULTS.type,
+	});
+
+	if (!type) {
+		return crash("An application type must be specified to continue.");
+	}
+
+	if (!Object.keys(templateMap).includes(type)) {
+		return crash(`Unknown application type provided: ${type}.`);
+	}
+
+	if (type === "webFramework") {
+		return selectFramework(args);
+	}
+
+	if (type === "remote-template") {
+		return processRemoteTemplate(args);
+	}
+
+	return templateMap[type];
+};
+
+export const selectFramework = async (args: Partial<C3Args>) => {
+	const frameworkMap = await getFrameworkMap();
+	const frameworkOptions = Object.entries(frameworkMap).map(
+		([key, config]) => ({
+			label: config.displayName,
+			value: key,
+		})
+	);
+
+	const framework = await processArgument<string>(args, "framework", {
+		type: "select",
+		label: "framework",
+		question: "Which development framework do you want to use?",
+		options: frameworkOptions,
+		defaultValue: C3_DEFAULTS.framework,
+	});
+
+	framework || crash("A framework must be selected to continue.");
```

Comment:
as before, can the framework not be defined here?

also, `framework || crash` is inconsistent with the type check in which we do have a standard `if`, I'd try to keep them consistent (and always use the `if` variant, but that's my personal preference)

Reply:
Same loophole as type, but I will make it consistent, since you asked :)